pylint/checkers/typecheck.py

```diff
@@ -1629,6 +1631,46 @@ def visit_unaryop(self, node):
             # Let the error customize its output.
             self.add_message("invalid-unary-operand-type", args=str(error), node=node)
 
+    @check_messages("unsupported-binary-operation")
+    def visit_binop(self, node: astroid.BinOp):
+        # Test alternative Union syntax PEP 604 - int | None
+        msg = "unsupported operand type(s) for |"
+        if node.op == "|" and (
+            not is_postponed_evaluation_enabled(node)
+            and isinstance(
+                node.parent, (astroid.AnnAssign, astroid.Arguments, astroid.FunctionDef)
+            )
+            or not PY310_PLUS
+            and isinstance(
+                node.parent,
+                (
+                    astroid.Assign,
+                    astroid.Call,
+                    astroid.Keyword,
+                    astroid.Dict,
+                    astroid.Tuple,
+                    astroid.Set,
+                    astroid.List,
+                ),
+            )
+        ):
+            for n in (node.left, node.right):
+                n = helpers.object_type(n)
+                if isinstance(n, astroid.ClassDef) and (
+                    n.name == "type"
+                    or any(
+                        [
+                            True
+                            for base in n.bases
+                            if isinstance(base, astroid.Name) and base.name == "type"
+                        ]
+                    )
```

Comment:
Maybe a simplification to do here ?
```suggestion
                    or any(
                        isinstance(base, astroid.Name) and base.name == "type" for base in n.bases
                    )
```

Reply:
Yeah, should have though after that sooner 😅